### PR TITLE
TASK: Add PgSQL optimization to NodeUriPathSegmentGenerator

### PR DESCRIPTION
**Pull request by dfeyer**

### CODE REVIEW

**Comment by dfeyer** [comment]
This one require https://github.com/neos/flow-development-collection/pull/1137

By using pgsql jsonb support to detect is an uriPathSegment is unique in the current path, we get a nice performance improvement

    without this change: ~100ms
    with this change: ~30ms

**Comment by dfeyer** [comment]
The change can be improve, but I'm intersted to have your first comments.

**Comment by dfeyer** [comment]
The last missing thing here, is to match only with current dimensions, but from my understanding the pgsql version works like the FlowQuery one.

**Comment by dfeyer** [comment]
If we can do the samething to improve node sorting index and the routing, we have a pretty good improvement for big site

File: Neos.Neos/Classes/Domain/Model/NodeUriPath.php [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+use Neos\ContentRepository\Exception\NodeException;
+use Neos\Eel\FlowQuery\FlowQuery;
+use Neos\Flow\Annotations as Flow;
+
+/**
+ * Node Uri Path
+ *
+ * @api
+ */
+class NodeUriPath
```
Comment (albe) [inline Neos.Neos/Classes/Domain/Model/NodeUriPath.php]: Anything against making this final?

File: Neos.Neos/Classes/Domain/Model/NodeUriPath.php [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+use Neos\ContentRepository\Domain\Model\NodeInterface;
+use Neos\ContentRepository\Domain\Repository\NodeDataRepository;
+use Neos\ContentRepository\Exception\NodeException;
+use Neos\Eel\FlowQuery\FlowQuery;
+use Neos\Flow\Annotations as Flow;
+
+/**
+ * Node Uri Path
+ *
+ * @api
```
Comment (albe) [inline Neos.Neos/Classes/Domain/Model/NodeUriPath.php]: Is this really public api?

Comment (dfeyer) [reply Neos.Neos/Classes/Domain/Model/NodeUriPath.php]: Not really, I can change this

**Pull request closed without merging**